src/console.rs

```diff
@@ -57,6 +60,74 @@ impl ConsoleLog {
     }
 }
 
+/// This is useful, but not implemented well. This needs to be either reworked and reimplemented, or utilised better.
+/// TODO: fix this
+pub trait SerializableConsoleOutput {
```

Comment:
And probably also this

Reply:
Ah, i see. You want ALL the traits to to just be in the sse-events. Okay lets see

Reply:
Yeah, since the stuff added in this branch isn't really used in demo management, it's all infrastructure for the sse broadcasting stuff, I think it should all go together in there. Rather than bloating this file with stuff that's only used for a specific purpose that's unrelated.